### PR TITLE
Retrying the docm workflow PR

### PR DESCRIPTION
**Pull request by chrisamiller**
Recreates https://github.com/genome/analysis-workflows/pull/591  after a series of adventures in git gone wrong.  Also addresses the Int/boolean issue, which was the last unresolved item from that PR.  

Still working on verifying this on a sample that may contain actual docm variants to be included, but that's running now

### CODE REVIEW

File: definitions/tools/docm_add_variants.cwl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+baseCommand: ["/usr/bin/java", "-Xmx8g", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "CombineVariants"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 9000
+      tmpdirMin: 25000
+    - class: DockerRequirement
+      dockerPull: mgibio/cle
+arguments:
+    ["-genotypeMergeOptions", "PRIORITIZE",
+     "--rod_priority_list", "callers,docm",
```
Comment (jasonwalker80) [inline definitions/tools/docm_add_variants.cwl]: Does this mean we lose the list of callers? I know in several projects where the number of callers, ex. 3 of 4, was used as a post-pipeline filter.

File: definitions/subworkflows/docm_cle.cwl [hunk 2]
```diff
@@ -37,24 +36,43 @@ steps:
...
+        in:
+            vcf: bgzip/bgzipped_file
+        out:
+            [indexed_vcf]
+    decompose:
+        run: ../tools/vt_decompose.cwl
+        in:
+            vcf: index/indexed_vcf
+        out:
+            [decomposed_vcf]
```
Comment (jasonwalker80) [inline definitions/subworkflows/docm_cle.cwl]: In the detect_variants.cwl workflow an index step was required. Is one not required here?

File: definitions/tools/docm_add_variants.cwl (hunk 1 above)
Comment (chrisamiller) [reply definitions/tools/docm_add_variants.cwl]: Nope!  see /gscmnt/gc2708/info/medseq/tmp/toiltest/output3/cromwell-executions/detect_variants.cwl/ab14a12b-b095-42e9-af4d-afa00cbca514/call-annotated_filter_index/execution/annotated_filtered.vcf.gz for an example

File: definitions/subworkflows/docm_cle.cwl (hunk 2 above)
Comment (chrisamiller) [reply definitions/subworkflows/docm_cle.cwl]: it's indexed again after the filter is applied, below.

File: definitions/tools/docm_add_variants.cwl (hunk 1 above)
Comment (jasonwalker80) [reply definitions/tools/docm_add_variants.cwl]: awesome! thanks!

Comment (chrisamiller) [reply definitions/tools/docm_add_variants.cwl]: to further expound - docm variants will never overwrite existing variants, only get added where there is no variant,  so the original line (with all caller info) will remain.

File: definitions/tools/filter_vcf_docm.cwl [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Filter variants from the DoCM detector"
+baseCommand: ["/usr/bin/perl", "/usr/bin/docm_filter.pl"]
+requirements:
+    - class: DockerRequirement
+      dockerPull: "mgibio/cle:v1.4.1"
+arguments: [
```
Comment (dufeiyu) [inline definitions/tools/filter_vcf_docm.cwl]: $docm_vcf, $normal_cram, $tumor_cram, $output_vcf_file, $set_filter_flag is the order of arguments that docm_filter.pl takes. Just wanna make sure that "docm_filtered_variants.vcf" will be in the position after tumor_bam and before filter_docm_variants flag.

**Comment by chrisamiller** [comment]
> Are those variants detected only by docm present ok in the final filtered vcf in your test ?
It depends.  They do appear in the docm output file, and then two of them make it through to the final filtered VCF (labelled as DOCM_ONLY  if `filter_docm_variants: true`). Several others are hard-filtered by other filters (gnomad, etc) and never make it to the very end.  If we set `filter_docm_variants: true`, they make it all the way through as PASS, again, unless they are removed by other filters.

File: definitions/tools/filter_vcf_docm.cwl (hunk 3 above)
Comment (chrisamiller) [reply definitions/tools/filter_vcf_docm.cwl]: Yep.  The negative numbers go before the `arguments` block, and positive numbers go after.

File: definitions/tools/docm_add_variants.cwl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     "--setKey","null",
```
Comment (dufeiyu) [inline definitions/tools/docm_add_variants.cwl]: Just wanna confirm that with this setting the variants only detected by docm will NOT have "set=" in INFO field

Comment (chrisamiller) [reply definitions/tools/docm_add_variants.cwl]: Correct. there won't be any "set", because they weren't detected by any caller.

**Pull request merged**